contracts/CG721.sol

```diff
@@ -1,40 +1,66 @@
-// SPDX-License-Identifier: MIT
+/// SPDX-License-Identifier: MIT
 pragma solidity ^0.8.15;
 
+
+/** @dev Imports the OpenZeppelin implemtations of ERC721, Enumerable extension for 
+  * inherited functions like totalSuppy(), Counters to automically generate the next token id after every mint, 
+  * and ownable
+  * for the onlyOwner modifer
+  */
 import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 
+/** @title CG 721
+  */  
 contract CG721 is ERC721Enumerable, Ownable {
   using Strings for uint256;
   using Counters for Counters.Counter;
   Counters.Counter private tokenIds;
-  uint256 public maxSupply = 20;
+  uint256 public maxSupply = 200;
   string public baseExtension = ".json";
-  string baseURI = '';
+  string baseURI = 'https://gateway.pinata.cloud/ipfs/QmTd3uKQp42cQ4dt2ztJNJVgJSnjQ36iYzKzwhuj9S1vD2/';
 
     constructor() ERC721("CG NFTs", "CG-NFT") {
     }
 
-    // Only owner to set IPFS base
-    function setBaseURI(string memory _newBaseURI) public onlyOwner {
+    /** @dev Sets the base URI for the generated Token URI
+      * @param _newBaseURI URL of IPFS folder gateway of JSON Files
+      */
+    function setBaseURI(string memory _newBaseURI) external onlyOwner {
     baseURI = _newBaseURI;
   }
 
-    // Internal to set base oppose to changing original OpenZeppelin ERC71.sol
+    /** @dev internal function used by TokenURI to return the baseURI.
+      * @return baseURI the base URI
+      */
     function _baseURI() internal view virtual override returns (string memory) {
     return baseURI;
   }
 
-    function mint() public returns(uint256) {
-      require(maxSupply > tokenIds.current(), "All tokens has been minted!");
+
+    /** @dev Allows owner to mint a single nft to their own wallet,
+      * requires the supply to be greater than or equal to 200,
+      * _safemint() is used to mint to msg.sender which is owner, and the current tokenID
+      * @return currentTokenId The token ID
+      */
+    function mint() external onlyOwner returns(uint256) {
```

Comment:
Implementation is wrong. Try to test this token functionality.

Reply:
I did, seemed to be functional, but made it more inline with ERC721 _safemint standards

Reply:
Functionality
---------
![Screen Shot 2022-07-27 at 9 45 10 AM](https://user-images.githubusercontent.com/69489205/181264092-d413c27b-4913-45fa-be62-216f661e725f.png)

<img width="556" alt="Screen Shot 2022-07-27 at 9 00 29 AM" src="https://user-images.githubusercontent.com/69489205/181263980-922eb6b3-26d7-4a42-a279-4b711c3dadef.png">

![Screen Shot 2022-07-27 at 9 45 26 AM](https://user-images.githubusercontent.com/69489205/181264214-e4560d09-8b5d-4bd4-84f8-ab1d3cb556f1.png)

![Screen Shot 2022-07-27 at 9 54 44 AM](https://user-images.githubusercontent.com/69489205/181265103-f42c3adc-3695-4b9e-91a4-d51c74454019.png)

![Screen Shot 2022-07-27 at 9 53 41 AM](https://user-images.githubusercontent.com/69489205/181264965-75172ae2-7a37-48e2-a0dd-d8b68705f40e.png)





-------